qcodes/instrument_drivers/ZI/ZIUHFLI.py

```diff
@@ -970,16 +964,45 @@ def __init__(self, name: str, device_ID: str, **kwargs) -> None:
                                 val_mapping={'ON': 1, 'OFF': 0},
                                 vals=vals.Enum('ON', 'OFF') )
 
-            self.add_parameter('signal_output{}_enable'.format(sigout),
-                                label="Enable signal output's amplitude.",
-                                set_cmd=partial(self._sigout_setter,
-                                                sigout-1, 0,
-                                                outputampenable[sigout]),
-                                get_cmd=partial(self._sigout_getter,
-                                                sigout-1, 0,
-                                                outputampenable[sigout]),
-                                val_mapping={'ON': 1, 'OFF': 0},
-                                vals=vals.Enum('ON', 'OFF') )
+            if 'MF' in self.props['options']:
+                for modeout in range(1, 9):
+                    self.add_parameter('signal_output{}_amplitude{}'.format(sigout, modeout),
+                                       label='Signal output amplitude',
+                                       set_cmd=partial(self._sigout_setter,
+                                                       sigout - 1, 1, 'amplitudes', output_mode=modeout - 1),
+                                       get_cmd=partial(self._sigout_getter,
+                                                       sigout - 1, 1, 'amplitudes', output_mode=modeout - 1),
+                                       unit='V')
+
+                    self.add_parameter('signal_output{}_enable{}'.format(sigout, modeout),
+                                       label="Enable signal output's amplitude.",
+                                       set_cmd=partial(self._sigout_setter,
+                                                       sigout - 1, 0,
+                                                       'enables', output_mode=modeout - 1),
+                                       get_cmd=partial(self._sigout_getter,
+                                                       sigout - 1, 0,
+                                                       'enables', output_mode=modeout - 1),
+                                       val_mapping=create_on_off_val_mapping())
+            else:
+                self.add_parameter('signal_output{}_enable'.format(sigout),
+                                   label="Enable signal output's amplitude.",
```

Comment:
what is meant with "enabling an amplitude"? is it just enabling that output so that the signal goes out of the device, or is it something else?

Reply:
Yes, it´s just enable signal output, the actual output. I fixed the label and added docstring